### PR TITLE
XWIKI-5168: Don't allow some methods in velocity introspector

### PR DESCRIPTION
**Pull request by surli**
JIRA: https://jira.xwiki.org/browse/XWIKI-5168

### CODE REVIEW

File: xwiki-commons-core/xwiki-commons-velocity/src/test/java/org/xwiki/velocity/introspection/SecureIntrospectorTest.java [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+/**
+ * Tests for {@link SecureIntrospector}.
+ *
+ * @version $Id$
+ */
+public class SecureIntrospectorTest
```
Comment (vmassol) [inline xwiki-commons-core/xwiki-commons-velocity/src/test/java/org/xwiki/velocity/introspection/SecureIntrospectorTest.java:37]: public keyword can be removed.

File: xwiki-commons-core/xwiki-commons-velocity/src/test/java/org/xwiki/velocity/introspection/SecureIntrospectorTest.java [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+{
+    @Mock
+    private Logger logger;
+
+    class CustomFile extends File {
```
Comment (vmassol) [inline xwiki-commons-core/xwiki-commons-velocity/src/test/java/org/xwiki/velocity/introspection/SecureIntrospectorTest.java:42]: `{` is supposed to be on a new line in our code rules.

File: xwiki-commons-core/xwiki-commons-velocity/src/main/checkstyle/checkstyle-suppressions.xml [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+-->
+
+<!DOCTYPE suppressions PUBLIC
+    "-//Puppy Crawl//DTD Suppressions 1.0//EN"
+    "http://www.puppycrawl.com/dtds/suppressions_1_0.dtd">
+
+<suppressions>
+  <!-- The default whitelists of accepted method are too long for this check -->
+  <suppress checks="ExecutableStatementCount" files="SecureIntrospector.java"/>
```
Comment (vmassol) [inline xwiki-commons-core/xwiki-commons-velocity/src/main/checkstyle/checkstyle-suppressions.xml:29]: What about constructing the list in a single statement?

Comment (surli) [reply xwiki-commons-core/xwiki-commons-velocity/src/main/checkstyle/checkstyle-suppressions.xml:29]: > What about constructing the list in a single statement?

Hmmm I could indeed probably use an `Arrays.asList` here. Will check that.

**Pull request merged**